modules/core/shared/src/main/scala/io/circe/JsonObject.scala

```diff
@@ -166,6 +166,21 @@ sealed abstract class JsonObject extends Serializable {
    */
   final def filterKeys(pred: String => Boolean): JsonObject = filter(field => pred(field._1))
 
+  /**
+   * Perform a deep merge of this JSON object with another JSON object.
+   *
+   * Objects are merged by key, values from the argument JSON take
+   * precedence over values from this JSON. Nested objects are
+   * recursed.
```

Comment:
Just thought I should maybe link to `Json.deepMerge` here to explain for the behaviour of values being merged?

Reply:
Sounds good to me. I can never remember the exact syntax for linking methods in API docs—if you can't find it easily `[[Json]]` would definitely be fine.

Reply:
I think what I've added is the right syntax... At least IntelliJ knows what's going on!

Reply:
Oh, that was easier than I thought!